src/handler.rs

```diff
@@ -154,12 +156,19 @@ impl RequestHandler {
 
                     Ok(resp)
                 }
-                Err(status) => error_page::error_response(
-                    method,
-                    &status,
-                    &self.opts.page404,
-                    &self.opts.page50x,
-                ),
+                Err(status) => {
+                    if let Some(response) =
```

Comment:
I would like to see the `page_fallback` option check inside the `handler.rs` rather than within the `fallback_response()`.
You can guide you from this server option.

```rs
            // `Basic` HTTP Authorization Schema
            if !self.opts.basic_auth.is_empty() {
               // ....
```

Even you could pass just a string and check if empty to skip `fallback_response()` otherwise process the `page_fallback` string value inside `fallback_response()`.

Reply:
I have pushed a commit to implement your changes. Please let me know if you have any further suggestions.